shared/utils/domains.ts

```diff
@@ -1,69 +1,54 @@
+import invariant from "invariant";
 import { trim } from "lodash";
+import env from "../env";
 
 type Domain = {
-  tld: string;
   subdomain: string;
   domain: string;
+  custom?: boolean;
 };
 
+// strips protocol and whitespace from input
+// then strips the path and query string
+function normalizeUrl(url: string) {
+  return trim(url.replace(/(https?:)?\/\//, "")).split(/[/:?]/)[0];
+}
+
 // we originally used the parse-domain npm module however this includes
 // a large list of possible TLD's which increase the size of the bundle
 // unnecessarily for our usecase of trusted input.
-export function parseDomain(url?: string): Domain | null | undefined {
-  if (typeof url !== "string") {
-    return null;
-  }
-  if (url === "") {
+
+// TODO: make this private
+// this only applies to internal domains (hosted domains)
+export function parseDomain(url: string): Domain | null | undefined {
+  if (!url) {
```

Comment:
`""` is falsey, and so is `undefined`. Other than `undefined`, the input should never be a non-string datatype according to the function typing